dashboard/modules/serve/serve_rest_api_impl.py

```diff
@@ -226,6 +226,24 @@ def log_config_change_default_warning(self, config):
                     )
                     break
 
+            for deployment in [
+                d for app in config.applications for d in app.deployments
```

Comment:
Non-blocker (nit): For readabilities, it's better to just write this as 2 for loops. But logic-wide it's equivalent 🙃
```
for app in config.applications:
    for deployment in app.deployments:
        ....
```

Reply:
In this case I think we can just call return 🙃

Reply:
We have two loops to go through though 😭

Reply:
Yea, return will kill both loops, since this is the last thing in this function. but again it's a bit nit, so feel free to keep as is

Reply:
deeply-nested early returning is pretty error prone (easy for people to not notice this and introduce subtle bugs in the future)